core/src/main/java/org/apache/accumulo/core/clientImpl/TabletServerBatchReaderIterator.java

```diff
@@ -407,7 +408,8 @@ public void run() {
             failures.putAll(tsFailures);
           }
         }
-
+      } catch (EOFException e) {
+        fatalException = e;
```

Comment:
Eh, this is really just one of many places where the code may not handle the error thrown from the RPC layer well. I don't think it uniquely stands out. I only noticed the MetadataLocationObtainer one because it receives the new EOFException, but then just treats it the same as other IOExceptions.